locallib.php

```diff
@@ -799,3 +897,30 @@ function zoom_get_alternative_host_array_from_string($alternativehoststring) {
     $alternativehostarray = array_filter(explode(',', $alternativehoststring));
     return $alternativehostarray;
 }
+
+/**
+ * Creates an iCalendar_event for a Zoom meeting.
+ *
+ * @param stdClass $event The meeting object.
+ * @param string $description The event description.
+ *
+ * @return iCalendar_event
+ */
+function zoom_helper_icalendar_event($event, $description) {
+    global $CFG;
+
+    // Match Moodle's uid format for iCal events.
+    $hostaddress = str_replace('http://', '', $CFG->wwwroot);
+    $hostaddress = str_replace('https://', '', $hostaddress);
+    $uid = $event->id . '@' .$hostaddress;
```

Comment:
My fault. Looks like I forgot a space here.
```suggestion
    $uid = $event->id . '@' . $hostaddress;
```

Reply:
Thanks for picking that up and your extensive review @jrchamp 

I have updated the space issue in locallib.php